File: quantinium/(200)_Notes/University/DSA/code/bubble_sort.c
#include <stdio.h>

void swap(int *x, int *y) {
    int temp = *x;
    *x = *y;
    *y = temp;
}

void bubble_sort(int arr[10], int size) {
    for (int i = 0; i < size; i++) {
        for (int j = 0; j < size - i - 1; j++) {
            if (arr[j] > arr[j + 1]) {
                swap(&arr[j], &arr[j + 1]);
            }
        }
    }
}

int main() {
    int arr[10] = {12, 2, 28, 47, 1, 89, 27, 37, 9, 8};
    size_t size = sizeof(arr) / sizeof(arr[0]);

    bubble_sort(arr, size);
    for (int i = 0; i < size; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

File: quantinium/(200)_Notes/University/DSA/code/bubble_sort.py
arr = [12, 2, 28, 47, 1, 89, 27, 37, 9, 8]

for i in range(len(arr)):
    for j in range(len(arr) - i - 1):
        if arr[j] > arr[j + 1]:
            arr[j],arr[j + 1] = arr[j + 1], arr[j]

print(arr)
